ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -856,13 +856,12 @@ public void onCompletion(final GetBlobResult routerResult, Exception routerExcep
                   if (securityException == null) {
                     if (subResource != null) {
                       BlobInfo blobInfo = routerResult.getBlobInfo();
-                      Map<String, String> userMetadata = RestUtils.buildUserMetadata(blobInfo.getUserMetadata());
-                      if (userMetadata == null) {
+                      if (RestUtils.getBooleanHeader(restRequest.getArgs(), SEND_USER_METADATA_AS_RESPONSE_BODY,
```

Comment:
since we save this as a boolean object, we might want to just get the value directly instead of using getBooleanHeader, so that we don't have to do a `toString()`

Reply:
Changed !